src/upper-level-components/GoogleLoginScreen.jsx

```diff
@@ -0,0 +1,104 @@
+import BraveLogoHorizontalSlate from '../pdf/BraveLogoHorizontalSlate.png'
+
+function GoogleLoginScreen(props) {
+  const { onLogin } = props
+  const [cookies, setCookie, removeCookie] = useCookies(['googleIdToken'])
+  const [errorMessage, setErrorMessage] = useState('Waiting for user login.')
+
+  async function handleGoogleIdToken(googleIdToken) {
+    try {
+      const res = await axios.post(`${process.env.REACT_APP_PA_GOOGLE_API}/pa/get-google-payload`, { googleIdToken })
+
+      // throw Error (also remove cookie for Google ID token) in the event of failure
+      /* NOTE: there are too many fields in the Google payload to check for each;
+       * expect that if res.data exists, it is properly populated. */
+      if (res.status !== 200 || !res.data) {
+        throw new Error('There was a problem interacting with the PA Google API.')
+      }
+
+      const googlePayload = res.data
+
+      // run login function while providing only the email and name of the account
+      onLogin({ email: googlePayload.email, name: googlePayload.name })
+    } catch (error) {
+      setErrorMessage(error.message)
+      removeCookie('googleIdToken') // ensure cookie is undefined
+    }
+  }
+
+  const googleLogin = useGoogleLogin({
+    /* Use the Google authorization code flow; this means that a separate OAuth2 client
+     * will get tokens from Google using a received Google authorization code. */
+    flow: 'auth-code',
+    // upon successful log in of a Google account and the receival of an authorization code (response.code)
+    onSuccess: async response => {
+      const googleAuthCode = response.code
+
+      try {
+        const res = await axios.post(`${process.env.REACT_APP_PA_GOOGLE_API}/pa/get-google-tokens`, { googleAuthCode })
+
+        // ensure that response was successful and returned all tokens needed
+        if (res.status !== 200 || !res.data || !res.data.googleAccessToken || !res.data.googleIdToken) {
+          throw new Error('There was a problem interacting with the PA Google API.')
+        }
+
+        // googleTokens contains members googleAccessToken and googleIdToken
+        /* NOTE: at present, googleAccessToken is not used, and is discarded after this function completes.
+         * For further interaction with Google's APIs, googleAccessToken will become useful. */
+        const googleTokens = res.data
+
+        setCookie('googleIdToken', googleTokens.googleIdToken)
+        handleGoogleIdToken(googleTokens.googleIdToken)
+      } catch (error) {
+        setErrorMessage(error.message)
+        removeCookie('googleIdToken') // unsure cookie is undefined
+      }
+    },
+    onError: error => setErrorMessage(error.message),
+    // see https://developers.google.com/identity/protocols/oauth2/scopes#google-sign-in
+    scope: 'email openid profile',
+  })
+
+  // attempt to log in if a Google ID token exists in cookies
+  if (cookies.googleIdToken !== undefined) {
+    handleGoogleIdToken(cookies.googleIdToken)
```

Comment:
I chose to let the handleGoogleIdToken function run asynchronously as I'm not sure if a React component can return a Promise... what do you think?

At present, given a cookie already exists, the handleGoogleIdToken function starts running, and then an empty div element is returned. When handleGoogleIdToken completes, it will update one of the component's states, or the googleIdToken cookie, after which the entire component will refresh.

Reply:
Ohhh I see what you mean, I'm thinking it might be better to use a useEffect hook in this case. My understanding is that in React useEffect is used to handle "side effects" like fetching data. So I'm thinking something like this
``` useEffect(() => {
    const loginWithToken = async () => {
      if (cookies.googleIdToken !== undefined) {
        await handleGoogleIdToken(cookies.googleIdToken);
      }
    };

    loginWithToken();
  }, [cookies.googleIdToken]); // This dependency will ensure this re-runs if the googleIdToken changes ```
  
So the useEffect will run when the component renders, and then run again if the googleIdToken changes. What do you think?

Reply:
Hmm, upon doing a little research, I cannot use `useEffect` with a cookie being in the dependency list, it must be a state created with `useState`. I can add a `googleIdToken` state, but I feel that is unnecessary overhead when a cookie already exists... I'll try and wrap the existing code in an effect, but I'm a little worried that the login screen will appear for a split second when googleIdToken exists in the cookies. This is why I am returning an empty div in the existing check.

Reply:
OK, I've implemented a solution that I'm happy with, and it uses effects. Let me know your thoughts!

Reply:
Ahh shoot, I was thinking the useCookies worked similarly to useState but I see what you mean, and I agree it doesn't make sense to add `googleIDToken` into state on top of it existing as a cookie. Hopefully wrapping it still works.